subsys/mgmt/ec_host_cmd/ec_host_cmd_handler.c

```diff
@@ -231,6 +222,13 @@ static void ec_host_cmd_thread(void *hc_handle, void *arg2, void *arg3)
 	}
 }
 
+#ifndef CONFIG_EC_HOST_CMD_DEDICATED_THREAD
+void ec_host_cmd_task(void)
+{
+	ec_host_cmd_thread(&ec_host_cmd, NULL, NULL);
```

Comment:
You should add a call to `k_thread_abort()` if `ec_host_cmd_thread()` returns.  This is done for normal [thread entry](https://github.com/zephyrproject-rtos/zephyr/blob/442a4529031e1a545e9c3ec254235b8050ad58f6/lib/os/thread_entry.c#L38).

You should also add `FUNC_NORETURN` to the protoype and declaration of `ec_host_cmd_task`

Reply:
You're correct that if the thread entry that calls `ec_host_cmd_task` exits, `k_thread_abort()` will get called.

I mistakenly thought `ec_host_cmd_thread()` had a path that can exit, so adding the `k_thread_abort()` would ensure that `ec_host_cmd_task()` can't return.

I think you should add the FUNC_NORETURN attribute to both `ec_host_cmd_task()` and `ec_host_cmd_thread()`.  This ensures someone doesn't add accidentally add an error path that returns from these routines.

Reply:
Makes sense.